tests/components/roomba/test_config_flow.py

```diff
@@ -0,0 +1,159 @@
+
+VALID_YAML_CONFIG = {
+    CONF_HOST: "1.2.3.4",
+    CONF_BLID: "blid",
+    CONF_PASSWORD: "password",
+    CONF_CERT: "/etc/ssl/certs/ca-certificates.crt",
+    CONF_CONTINUOUS: True,
+    CONF_DELAY: 1,
+}
+
+
+def _create_mocked_roomba(
+    roomba_connected=None, master_state=None, connect=None, disconnect=None
+):
+    mocked_roomba = MagicMock()
+    type(mocked_roomba).roomba_connected = PropertyMock(return_value=roomba_connected)
+    type(mocked_roomba).master_state = PropertyMock(return_value=master_state)
+    type(mocked_roomba).connect = MagicMock(side_effect=connect)
+    type(mocked_roomba).disconnect = MagicMock(side_effect=disconnect)
+    return mocked_roomba
+
+
+async def test_form(hass):
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["errors"] == {}
+
+    mocked_roomba = _create_mocked_roomba(
+        roomba_connected=True,
+        master_state={"state": {"reported": {"name": "myroomba"}}},
+    )
+
+    with patch(
+        "homeassistant.components.roomba.config_flow.Roomba",
+        return_value=mocked_roomba,
+    ), patch(
+        "homeassistant.components.roomba.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.roomba.async_setup_entry", return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], VALID_CONFIG,
+        )
+
+    assert result2["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result2["title"] == "myroomba"
+
+    assert result2["result"].unique_id == "blid"
+    assert result2["data"] == {
+        CONF_BLID: "blid",
+        CONF_CERT: "/etc/ssl/certs/ca-certificates.crt",
+        CONF_CONTINUOUS: True,
+        CONF_DELAY: 1,
+        CONF_HOST: "1.2.3.4",
+        CONF_PASSWORD: "password",
+    }
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_cannot_connect(hass):
+    """Test we handle cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    mocked_roomba = _create_mocked_roomba(
+        connect=RoombaConnectionError,
+        roomba_connected=True,
+        master_state={"state": {"reported": {"name": "myroomba"}}},
+    )
+
+    with patch(
+        "homeassistant.components.roomba.config_flow.Roomba",
+        return_value=mocked_roomba,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], VALID_CONFIG,
+        )
+
+    assert result2["type"] == data_entry_flow.RESULT_TYPE_FORM
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5343767)